### PR TITLE
add aws ses support

### PR DESCRIPTION
**Pull request by southeo**
- Adds email service
- Adds NOTIFICATION_FAILED job state. This is because if the job succeeds, but the email server fails, we don't want to re-run the job. 
- If job has already been completed (based on the hashed params), we notify the user right away. If the email fails, we send the job to the queue so we don't alter the previous job record. 
    - Maybe we could just immediately create a job record with the NOTIFICATION_FAILED state?

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -93,6 +100,24 @@
       <groupId>org.springframework.boot</groupId>
       <artifactId>spring-boot-starter-web</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.springframework</groupId>
+      <artifactId>spring-context-support</artifactId>
+      <version>6.1.5</version>
```
Comment (samleeflang) [inline pom.xml:106]: Can you put this in properties

File: pom.xml [hunk 2]
```diff
@@ -48,6 +48,13 @@
         <scope>import</scope>
         <type>pom</type>
       </dependency>
+      <dependency>
+        <groupId>software.amazon.awssdk</groupId>
+        <artifactId>bom</artifactId>
+        <version>2.28.21</version>
```
Comment (samleeflang) [inline pom.xml:54]: Put version on top in properties

File: pom.xml [hunk 3]
```diff
@@ -93,6 +100,24 @@
...
+      <artifactId>spring-boot-starter-mail</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>software.amazon.awssdk</groupId>
+      <artifactId>sesv2</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>software.amazon.awssdk</groupId>
+      <artifactId>ec2</artifactId>
```
Comment (samleeflang) [inline pom.xml:119]: Is this necessary?

File: pom.xml [hunk 4]
```diff
@@ -93,6 +100,24 @@
...
       <artifactId>spring-boot-starter-web</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.springframework</groupId>
+      <artifactId>spring-context-support</artifactId>
+      <version>6.1.5</version>
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-mail</artifactId>
```
Comment (samleeflang) [inline pom.xml:110]: Do we still use spring-mail?

File: src/main/java/eu/dissco/dataexporter/controller/DataExporterController.java [hunk 5]
```diff
@@ -31,17 +34,24 @@ public ResponseEntity<Void> scheduleJob(Authentication authentication,
...
   }
 
-  @PostMapping("/internal/{id}/running")
-  public ResponseEntity<Void> markJobAsRunning(@PathVariable("id") UUID id) {
-    service.markJobAsRunning(id);
+  @PostMapping("/internal/{id}/{jobState}")
+  public ResponseEntity<Void> updateJobState(@PathVariable("id") UUID id,
+      @PathVariable("jobState") String jobStateStr) throws InvalidRequestException {
+    service.updateJobState(id, getJobState(jobStateStr));
     log.info("Successfully marked job {} as running", id);
```
Comment (samleeflang) [inline src/main/java/eu/dissco/dataexporter/controller/DataExporterController.java:46]: I think the message needs an update as well to include the jobState

File: src/main/java/eu/dissco/dataexporter/repository/DataExporterRepository.java [hunk 6]
```diff
@@ -36,18 +39,41 @@ public void addJobToQueue(ExportJob job) throws InvalidRequestException {
...
         .set(EXPORT_QUEUE.TIME_STARTED, Instant.now())
         .where(EXPORT_QUEUE.ID.eq(id))
         .execute();
   }
 
+  public void markJobAsComplete(JobResult jobResult, JobState jobState) {
+    context.update(EXPORT_QUEUE)
+        .set(EXPORT_QUEUE.JOB_STATE, jobState)
+        .set(EXPORT_QUEUE.TIME_COMPLETED, Instant.now())
+        .set(EXPORT_QUEUE.S3_LINK, jobResult.s3Link())
```
Comment (samleeflang) [inline src/main/java/eu/dissco/dataexporter/repository/DataExporterRepository.java:61]: Starting to doubt if we need to call it S3_Link or just exporter_result or exporter_link to keep it agnostic. We might switched away from AWS then we are stuck with this name

File: src/test/resources/db/migration/V1__init_db.sql [hunk 7]
```diff
@@ -14,6 +14,7 @@ create table export_queue
     time_completed    timestamp with time zone,
     export_type       export_type              not null,
     hashed_params     uuid                     not null,
-    destination_email text                     not null
+    destination_email text                     not null,
+    s3_link           text
```
Comment (samleeflang) [inline src/test/resources/db/migration/V1__init_db.sql:18]: Better to make this generic

File: src/test/java/eu/dissco/dataexporter/controller/DataExporterControllerTest.java [hunk 8]
```diff
@@ -47,14 +50,40 @@ void testScheduleJob() throws Exception {
...
+
+  @Test
+  void testJobStateInvalid() {
+    // When / then
+    assertThrows(InvalidRequestException.class, () -> controller.updateJobState(ID, "invalid"));
+  }
+
+  @Test
+  void testMarkJobComplete() {
+    // Given
```
Comment (samleeflang) [inline src/test/java/eu/dissco/dataexporter/controller/DataExporterControllerTest.java:83]: Maybe exclude Given here as you do with the others

**Review by samleeflang — COMMENTED** [review]
Couple comments but overall looks good!

File: pom.xml (hunk 4 above)
Comment (southeo) [reply pom.xml:110]: no, removed

File: pom.xml (hunk 3 above)
Comment (southeo) [reply pom.xml:119]: no, removed

**Review by samleeflang — APPROVED** [review]
:gear:

**Pull request merged**